lib/Driver/DarwinToolChains.cpp

```diff
@@ -588,8 +588,66 @@ bool toolchains::Darwin::shouldStoreInvocationInDebugInfo() const {
 void 
 toolchains::Darwin::validateArguments(DiagnosticEngine &diags,
                                       const llvm::opt::ArgList &args) const {
+  // Validating bridging header files.
+  validateBridgingHeaderArgs(diags, args);
+
+  // Validating apple platforms deployment targets.
+  validateDeploymentTarget(diags, args);
+    
   // Validating darwin unsupported -static-stdlib argument.
   if (args.hasArg(options::OPT_static_stdlib)) {
     diags.diagnose(SourceLoc(), diag::error_darwin_static_stdlib_not_supported);
   }
 }
+
+void
+toolchains::Darwin::validateDeploymentTarget(DiagnosticEngine &diags,
+                                             const llvm::opt::ArgList &args) const {
+  const Arg *A = args.getLastArg(options::OPT_target);
+  if (!A)
+    return;
+  
+  // Check minimum supported OS versions.
+  llvm::Triple triple(llvm::Triple::normalize(A->getValue()));
```

Comment:
Now that this is in the toolchain you can actually do better: `getTriple()` gets the already-computed triple.